pkg/typesystem/typesystem.go

```diff
@@ -22,6 +27,17 @@ var (
 	ErrInvalidUsersetRewrite = errors.New("invalid userset rewrite definition")
 )
 
+// ContextWithTypesystem attaches the provided TypeSystem to the parent context.
+func ContextWithTypesystem(parent context.Context, typesys *TypeSystem) context.Context {
```

Comment:
We may consider using this more elsewhere now that it exists. Instead of passing around the TypeSystem everywhere we can inject it into the context and extract it where needed.